### ISSUE TITLE
TabRouter uses inner action before outer routeName has been found

### ISSUE THREAD

**Issue by monteithpj** [react-community/react-navigation#875]
I have a case where I have a TabRouter with nested StackRouters inside and when I send an action with an inner action the inner action is unpacked and sent to the children before it's reached the target of the outer action. eg, given Navigator setup:

```
MyNav = TabNavigator({
   A: StackNavigator({
      initialScreen: Screen,
      B: StackNavigator({...})
  })
})
```
when sending action:
```
{type: NAVIGATE, routeName: "B", action: SOME_INNER_ACTION}
```

MyNav sees that its current child is a router, so it delegates the action - however it delegates the inner action, even though it's current child is A, not B (the target of the outer action).

This is problematic when B isn't the current child of A, ie A is in it's initial state. The intention is that the outer action causes A to select B as the current child and delegate, but A has only been given the inner action, thus it won't select B at all.

I propose that when delegating to the current tab, the outer action should be sent unless the current tab is itself the target of the outer action.

If people agree that this is unintended behaviour, i'm happy to stick a PR up

EDIT: Corrected failure case

**Comment by kelset** [comment]
Pinging OP @monteithpj  since this issue is quite old and related to an old version of the lib. If no answer I'll close in 7 days.

**Comment by monteithpj** [comment]
@kelset This is still a problem - would be great to get this looked at =)

**Comment by kelset** [comment]
Hi @monteithpj, can you let me know if this PR #2674 fixed the issue? They seem to be somewhat related. (you can test it out via pointing to master, just be careful 'cause there are some minor BC)

**Comment by monteithpj** [comment]
Yup, this problem is still happening

**Comment by kelset** [comment]
@monteithpj we finally merged #895, can you try pointing to master to see if it's finally fixed?

**Comment by monteithpj** [comment]
Yes, this is working correctly now, thanks!
